Draft proposals: Project Constitution, Project Social Contract (#17)
jellyfin/jellyfin-meta: policies-and-procedures/jellyfin-constitution.md

# Jellyfin Consitution

Draft 1 of Version 1.0, 2021-02

> Based on the Debian Constitution, available at https://www.debian.org/devel/constitution

Reviewer: ```suggestion
Based on the [Debian Constitution](https://www.debian.org/devel/constitution).
```

## General Organizational Structure and Principles

> 1. The Jellyfin Project is organized into a heirarchical structure with the following layers, which shall be defined in more detail below:

Reviewer 2: ```suggestion
1. The Jellyfin Project is organized into a hierarchical structure with the following layers, which shall be defined in more detail below:
```

    i. The Project Leader

    ii. The Leadership Team

    iii. The Financial Committee

    iv. Subproject Leaders

    v. Technical Committees

    vi. The Contributor Function Teams

    vii. The Contributor Team as a whole

2. In the list above, a person or body is usually listed before any people or bodies whose decisions they can overrule or who they (help) appoint - but not everyone listed earlier can overrule everyone listed later.

3. Nothing in this constitution imposes an obligation on anyone to do work for the Project. A person who does not want to do a task which has been delegated or assigned to them does not need to do it. However, they must not actively work against these rules and decisions properly made under them.

4. A person may leave the Project or resign from a particular post they hold, at any time, by stating so publicly.

5. A person may hold several posts at one time, and be part of as many teams as desired.

## Role Definitions and Powers

### Project Leader

#### Powers

> 1. The Project Leader is reposible for the overall management of the project direction and releases. They are bound by the terms of the Jellyfin Social Contract as much as any other member, and embody the spirit and goals of this social contract.

Reviewer 2: ```suggestion
1. The Project Leader is responsible for the overall management of the project direction and releases. They are bound by the terms of the Jellyfin Social Contract as much as any other member, and embody the spirit and goals of this social contract.
```

2. The Project Leader is the final arbiter in any disputes or decisions raised from lower teams and persons where such a decision cannot be reached by more collaborative means within the teams themselves.

3. The Project Leader shall generally take a hands-off approach to day-to-day project decisions, delegating responsibility for these to the lower teams and persons.

4. The Project Leader shall also function as Release Manager for the core server, handling the coordination and creation of new releases of the Jellyfin server software. Creation of releases for clients or subcomponents is delegated to their respective Function Teams.

5. The Project Leader must always be a member of both the Leadership Team and the Financial Committee.

#### Procedure

1. The first Project Leader (Joshua Boniface) was and is self-appointed, as the original driver of the fork from Emby.

2. The Project Leader shall serve indefinitely at their discretion, but may resign at any time.

> 3. Upon resignation or indisposal of the Project Leader, one or more candidates for the position shall be proposed by the Leadership Team. A vote shall then be held by all members of the Jellyfin Contrbutor Team to confirm or select the chosen candidate(s). The method of election shall be ranked voting instant runoff until one candidate has secured an absolute majority of the vote. In the event that only a single candidate is proposed, and this candidate does not gain a majority of votes amongst the Contributor Team, the outgoing Project Leader shall cast a deciding vote to confirm or deny the candidate outright. If the outgoing Project Leader does not confirm the candidate, the process shall begin again and the previous candidate is inelligible for further consideration.

Reviewer 3: ```suggestion
3. Upon resignation or indisposal of the Project Leader, one or more candidates for the position shall be proposed by the Leadership Team. A vote shall then be held by all members of the Jellyfin Contributor Team to confirm or select the chosen candidate(s). The method of election shall be ranked voting instant runoff until one candidate has secured an absolute majority of the vote. In the event that only a single candidate is proposed, and this candidate does not gain a majority of votes amongst the Contributor Team, the outgoing Project Leader shall cast a deciding vote to confirm or deny the candidate outright. If the outgoing Project Leader does not confirm the candidate, the process shall begin again and the previous candidate is inelligible for further consideration.
```

4. The Project Leader may at their discretion appoint a temporary Interim Project Leader to handle their duties due to short-term absense or for the duration of a Projet Leader election.

### Leadership Team ("Core")

#### Powers

1. The Leadership Team is responsible for the day-to-day operations of the project and the handling of technical and managerial decisions for the project.

2. The Leadership Team is the only group with "Merge" permissions for the core server repositories, and are responsible for the merging of reviewed and approved Pull Requests into those repositories. All code changes must thus require implicit or explicit approval of a member of the Leadership Team.

> 3. The Leadership Team should at all times provide a unified front for decisions, working under the principle of democratic centralism - lively debate and disagreement within the team is encouraged during decisionmaking, but decisions should be accepted and promulaged by the team as a unified entity without public disagreement. If internal decisionmaking fails, the decision shall fall to the Project Leader instead.

Reviewer 2: ```suggestion
3. The Leadership Team should at all times provide a unified front for decisions, working under the principle of democratic centralism - lively debate and disagreement within the team is encouraged during decision-making, but decisions should be accepted and promulgated by the team as a unified entity without public disagreement. If internal decision-making fails, the decision shall fall to the Project Leader instead.
```

> 3. The Leadership Team should at all times provide a unified front for decisions, working under the principle of democratic centralism - lively debate and disagreement within the team is encouraged during decisionmaking, but decisions should be accepted and promulaged by the team as a unified entity without public disagreement. If internal decisionmaking fails, the decision shall fall to the Project Leader instead.

Reviewer 4: ```suggestion
3. The Leadership Team should at all times provide a unified front for decisions, working under the principle of democratic centralism - lively debate and disagreement within the team is encouraged during decisionmaking, but decisions should be accepted and promulaged by the team as an unified entity without public disagreement. If internal decisionmaking fails, the decision shall fall to the Project Leader instead.
```

Reviewer: The original is correct. The `u` in "unified" is pronounced `/ju:/` and `/j/` is a consonant sound, which means `a unified` is the correct form here.

One of those weird English exceptions :p

4. Members of the Leadership Team shall be removed from their position after 1 year of inactivity with the project, subject to discussion by the remaining Leadership Team members.

### Financial Committee

#### Powers

1. The Financial Committee is a sub-group of the Leadership Team who is in charge of allocating funds from the OpenCollective donation pool.

2. The Financial Committee shall decide which if any expenses are legitimate, as well as handle recurring expenses for the project such as infrastructure.

3. Any member of the Financial Committee shall have the power to approve or veto any expense. If a veto is used, the reasoning must be explained to the Leadership Team.

#### Procedure

1. The initial and current Financial Committee consists of Joshua Boniface, Andrew Rabert, and Anthony Lavado.

2. The Financial Committee shall always include exactly 3 members to ensure transparency in all funds allocations.

3. New members shall be proposed by the existing Financial Committee should the need arise, by direct nomination by the departing member. The new member must be confirmed unanimously by the remaining members.

### Subproject Leaders

#### Powers

> 1. Subproject Leaders are those members of the team with a leadership role in guiding a particular subproject. While the Project Leader guides the project as a whole, Subproject Leaders guide individual subprojects in a similar fashion. A subproject is any component of the Jellyfin ecosystem, under the Jellyfin Organization, which is not a part of the core server itself. This includes clients, libraries for 3rd party projects, or other similar repositories for which the Project Leader delegates this responsbility.

Reviewer 2: ```suggestion
1. Subproject Leaders are those members of the team with a leadership role in guiding a particular subproject. While the Project Leader guides the project as a whole, Subproject Leaders guide individual subprojects in a similar fashion. A subproject is any component of the Jellyfin ecosystem, under the Jellyfin Organization, which is not a part of the core server itself. This includes clients, libraries for 3rd-party projects, or other similar repositories for which the Project Leader delegates this responsibility.
```

    ii. For subprojects that originate as the sole work of an existing Jellyfin Team member, that person shall become Subproject Leader at the time the project is transferred to the Jellyfin Organization.

   iii. For subprojects that originate as the sole work of an outside contributor who desires to make the project a subproject of the Jellyfin Organization, that person shall become a member of the Jellyfin Team, subject to standard nomination rules for Team Members, and shall become Subproject Leader at the time the project is transferred to the Jellyfin Organization. If nomination of the member fails for any reason, the project shall not be integrated into the Jellyfin Organization and shall remain a 3rd-party project.

    iv. For subprojects that, at the time of the adoption of this consitution, do not already have a clear Subproject Leader, elections shall be held by the existing contributors to select the Subproject Leader. The method of election shall be ranked voting instant runoff until one candidate has secured an absolute majority of the vote.

> 2. There shall always be exactly one Subproject Leader for any given project.

Reviewer 2: How would this rule apply to the Android subproject specifically? Right now, both me and @nielsvanvelzen are admins of the GitHub repository, and we share the subproject leader role. I'd prefer to continue with that process, since managing the project and making decisions together has been very productive and I wouldn't even want to have the full authority myself.

Author: Very good point. Since you guys are already sharing it and it's already working out, and you both already have Admin, Co-Subproject Leaders (more like Andrew and I were in the early days) makes the most sense. You both have the powers but, ideally, come to consensus agreement on any action or decision. Does that work? /cc @nielsvanvelzen

Author: I have addressed this by allowing either one or two in this line. I actually didn't realize Android already had two :D

Reviewer 5: How I see it is that I already am a subproject leader for android overall (so androidtv, android mobile and kotlin/java apiclient). @Maxr1998 is the creator of the current mobiel app and maintains most of it so I consider him as the subproject leader for the android mobile app underneath the "android" project.

Illustration because I find it hard to explain in text:

- Jellyfin
  - server/web
  - Android | Niels
    - jellyfin-android (mobile) | Max
    - jellyfin-androidtv | Niels - With assistance from Bill
    - jellyfin-apiclient-java | Niels
  - ...other things

Reviewer 2: Thanks for the update abd the change, sounds good for me.

@nielsvanvelzen theoretically, I'd see the apiclient as independent from the Android subproject, and I'd even argue that (at least on paper) AndroidTV is an independent subproject from Android as well.
At least if we want to keep the hierarchy flat, because the structure you described sounds like having "sub-sub-projects".

Reviewer 5: I think those three projects are related to each other so it's easy to "group" them, both apps use the apiclient so contributors for the apps might contribute to the apiclient etc.

> <!-- markdownlint-disable -->

Reviewer 2: To only disable the duplicate headings warning:
```suggestion
<!-- markdownlint-disable MD024 -->
```

Or, even more specific, put this somewhere in the file (could also be at the end):
```
<!-- markdownlint-configure-file { "MD024": { "allow_different_nesting": true } } -->
```

Author: I just disabled them all because meh. It complains both about duplicate headers and numbering being explicit.

    ii. For sub-projects that originate as the sole work of an existing Jellyfin Team member, that person shall become Subproject Leader at the time the project is transferred to the Jellyfin Organization.

   iii. For sub-projects that originate as the sole work of an outside contributor who desires to make the project a sub-project of the Jellyfin Organization, that person shall become a member of the Jellyfin Team, subject to standard nomination rules for Team Members, and shall become Subproject Leader at the time the project is transferred to the Jellyfin Organization. If nomination of the member fails for any reason, the project shall not be integrated into the Jellyfin Organization and shall remain a 3rd-party project.

    iv. For sub-projects that, at the time of the adoption of this condition, do not already have a clear Subproject Leader, elections shall be held by the existing contributors to select the Subproject Leader. The method of election shall be ranked voting instant runoff until one candidate has secured an absolute majority of the vote.

> 2. There may be either one Subproject Leader, or two Subproject Co-Leaders for larger sub-projects, for any given sub-project. Co-Leaders shall share all powers but should strive for consesnsus decisions.

Reviewer 3: ```suggestion
2. There may be either one Subproject Leader, or two Subproject Co-Leaders for larger sub-projects, for any given sub-project. Co-Leaders shall share all powers but should strive for consensus in all decisions.
```

Author: Both work, in this case I'm using "consensus decisions" as a single noun, i.e. "decisions that are agreed on by consensus".

Reviewer 3: Oh I see. However, the typo is still there :)
